File: javascript/selectionSort.js
function findBiggest(arr) {
    let biggest = 0
    let biggestIndex = 0

    for (let i = 0; i < arr.length; i++) {
        if (biggest < arr[i]) {
            biggest = arr[i]
            biggestIndex = i
        }
    }
    return biggestIndex
}


function selectionSort(arr) {
    const resultArr = []
    const copiedArr = [...arr]

    for (let i = 0; i < arr.length; i++) {
        let biggestIndex = findBiggest(copiedArr)
        let biggest = copiedArr.splice(biggestIndex, 1)
        resultArr.push(...biggest)
    }

    return resultArr
}

console.log(selectionSort([4, 9, 12, 2, 0, 8]))

File: python/scripts/selection_sort.py
def find_smallest(arr):
    smallest = arr[0]
    smallest_index = 0
    for i in range(1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallest_index = i
    return smallest_index


def selection_sort(arr):
    result_arr = []
    copied_arr = arr.copy()

    for _ in range(len(copied_arr)):
        smallest_index = find_smallest(copied_arr)
        smallest = copied_arr.pop(smallest_index)
        result_arr.append(smallest)

    return result_arr
